libraries/standard/http/utest/http_utest.c

```diff
@@ -88,9 +101,96 @@ static _headers_t expectedHeaders = { 0 };
 static int testRangeStart = 0;
 static int testRangeEnd = 0;
 static uint8_t httpBuffer[ HTTP_TEST_BUFFER_LEN ] = { 0 };
+static const uint8_t * pValueLoc = NULL;
+static size_t valueLen = 0u;
+static HTTPResponse_t testResponse = { 0 };
+static const size_t headerFieldInRespLoc = 44;
+static const size_t headerFieldInRespLen = sizeof( "test-header1" ) - 1u;
+static const size_t headerValInRespLoc = 58;
+static const size_t headerValInRespLen = sizeof( "test-value1" ) - 1u;
+static http_parser * pCapturedParser = NULL;
+static http_parser_settings * pCapturedSettings = NULL;
+static const char * pExpectedBuffer = NULL;
+static size_t expectedBufferSize = 0u;
+static bool invokeHeaderFieldCallback = false;
+static const char * pFieldLocToReturn = NULL;
+static size_t fieldLenToReturn = 0u;
+static bool invokeHeaderValueCallback = false;
+static const char * pValueLocToReturn = NULL;
+static size_t valueLenToReturn = 0u;
+static int expectedValCbRetVal = 0;
+static bool invokeHeaderCompleteCallback = false;
+static unsigned int parserErrNo = 0;
 
 /* ============================ Helper Functions ============================== */
 
+void parserInitExpectationCb( http_parser * parser,
```

Comment:
I know these callbacks are for mocking, but could they documented as well for consistency with other definitions?

Reply:
Will add documentation